lib/bundler/cli.rb

```diff
@@ -407,6 +407,8 @@ def inject(name, version, *gems)
       "print the lockfile to STDOUT instead of writing to the file system"
     method_option "lockfile", :type => :string, :default => nil, :banner =>
       "the path the lockfile should be written to"
+    method_option "gems", :type => :array, :default => [], :banner =>
```

Comment:
what if we just turned the `--update` option into taking an array, and pass in `true` if that array is empty?